File: apps/ios/Barns/BarnsApp/Domain/UseCases/Patterns/GetPatternsUseCase.swift
/// Returns the local wall-greenery patterns.
struct GetPatternsUseCase {
    private let repository: PatternRepository

    init(repository: PatternRepository) {
        self.repository = repository
    }

    func execute() async throws -> [WallGreenPattern] {
        try await repository.wallGreenPatterns()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/patterns/GetPatternsUseCase.kt
package com.barns.app.domain.usecase.patterns

import com.barns.app.domain.model.WallGreenPattern
import com.barns.app.domain.repository.PatternRepository

/**
 * Returns the local wall-greenery patterns.
 */
class GetPatternsUseCase(
    private val repository: PatternRepository,
) {
    suspend fun execute(): List<WallGreenPattern> = repository.wallGreenPatterns()
}
